src/pydot/dot_parser.py

```diff
@@ -497,6 +497,9 @@ def graph_definition():
 def parse_dot_data(s):
     """Parse DOT description in (unicode) string `s`.
 
+    This function is NOT thread-safe due to the internal use of `pyparsing`.
+    Use a lock if needed.
+
     @return: Graphs that result from parsing.
     @rtype: `list` of `pydot.Dot`
     """
```

Comment:
Makes sense to me!

At some point we might think about reformatting all of the docstrings into Sphinx format — or, better, [sphinx one-line format](https://github.com/NilsJPWerner/autoDocstring/blob/c9da64126fd9e667decd9d85b4e5b53c60372ea7/docs/one-line-sphinx.md) (the difference being that Sphinx format uses separate `:param foo:` and `:type foo:` lines, while one-line format is `:param \<type> foo:`), then we could make these into "Warning" admonitions. e.g.:

```python
def parse_dot_data(s):
    """Parse DOT description in (unicode) string `s`.

    :return list[pydot.Dot]: Graphs that result from parsing.

    .. warning:: This function is NOT thread-safe due to
       the internal use of :mod:`pyparsing`.
       Use a :class:`threading.Lock` if needed.
    """
```

But it makes more sense to do that all at once, in a separate PR. IMHO.